lib/common/ipc_server.c

```diff
@@ -585,22 +594,58 @@ pcmk__ipc_prepare_iov(uint32_t request, const GString *message,
     iov[0].iov_base = header;
 
     header->version = PCMK__IPC_VERSION;
-    header->size = message->len + 1;
-    total = iov[0].iov_len + header->size;
+
+    /* We are passed an index, which is basically how many times this function
+     * has been called.  This is how we support multi-part IPC messages.  We
+     * need to convert that into an offset into the buffer that we want to start
+     * reading from.
+     *
+     * Each call to this function can send max_send_size, but this also includes
+     * the header and a null terminator character for the end of the payload.
+     * We need to subtract those out here.
+     */
+    offset = index * (max_send_size - iov[0].iov_len - 1);
```

Comment:
Since we use this calculation twice, we *could* store it in a variable like `max_chunk_size`. Efficiency boost is insignificant, but it might be clearer.